src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -64,6 +64,12 @@ internal class ConnectionManager : IConnectionManager {
             _userConnections = new ConcurrentDictionary<Uri, IConnection>(userConnections);
 
             UpdateRecentConnections();
+
+            _shell.Idle += OnIdle;
+        }
+
+        private void OnIdle(object sender, EventArgs e) {
```

Comment:
BTW, main thread does exist and can be reentered when DTE is still not ready. Perhaps instead of using DTE get dialog owner from IVsUiShell ?